Sources/ShapedArray/Operations/pack.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  File.swift
+//  
+//
+//  Created by Jaap Wijnen on 10/08/2022.
+//
+
+extension ShapedArray {
+    static func pack(_ values: [ShapedArray], axis: Int = 0) -> ShapedArray {
+
+        precondition(!values.isEmpty, "Cannot pack empty array of ShapedArrays.")
+        let shape = values.first!.shape
+        precondition(axis >= 0 && axis <= shape.count, "axis = \(axis) is not within [0, \(shape.count)]")
```

Comment:
Originally the range for axis could go negative, any reason not to support this?

https://github.com/s4tf/s4tf/blob/a31f79a7a8e92ff708b1034286a23510b23af5f3/Sources/x10/swift_bindings/apis/RawOpsManual.swift#L2183

Reply:
We don't have to support negative axes. S4TF does so because it's reflecting what the underlying backend does. We can implement it differently than TensorFlow.

For precedent, NumPy does not permit arbitrary negative axes. It [only allows `-1`](https://numpy.org/doc/stable/reference/generated/numpy.stack.html), which short-circuits to the last axis. In Swift, we could force you to explicitly specify the last axis. Using `<= shape.count` in the precondition instead of `<` permits this behavior.

TensorFlow does not allow specifying `axis=rank`, only `axis=rank-1`. If we delegate to a TensorFlow-like backend in the future, we can just subtract `rank` from the axis before passing it in. The public API can be different than what happens internally, as long as it produces the same result.

Reply:
I like eschewing negative for simplicity 👍